tests/foreman/cli/test_fact.py

```diff
@@ -61,3 +63,46 @@ def test_negative_list_by_name(module_target_sat):
     assert (
         module_target_sat.cli.Fact().list(options={'search': f'fact={gen_string("alpha")}'}) == []
     )
+
+
+@pytest.mark.no_containers
+@pytest.mark.rhel_ver_list([settings.content_host.default_rhel_version])
+def test_positive_update_client_facts(
+    module_target_sat, rhel_contenthost, module_org, module_location, module_activation_key
+):
+    """Update client facts and verify the facts are updated in Satellite.
+
+    :id: ea94ccb7-a125-4be3-932a-bfcb035d3604
+
+    :steps:
+        1. Add a new interface to the host.
+        2. Register the host to Satellite
+        3. Update the facts in Satellite.
+        4. Read all the facts for the host.
+        5. Verify that the facts are updated in Satellite.
+
+    :expectedresults: Facts are successfully updated in the Satellite.
+    """
+    mac_address = gen_mac(multicast=False)
+    ip = gen_ipaddr()
+    # Create eth1 interface on the host
+    add_interface_command = (
+        f'ip link add eth1 type dummy && ifconfig eth1 hw ether {mac_address} &&'
+        f'ip addr add {ip}/24 brd + dev eth1 label eth1:1 &&'
+        'ip link set dev eth1 up'
+    )
+    result = rhel_contenthost.execute(add_interface_command)
+    assert result.status == 0
+    result = rhel_contenthost.api_register(
+        module_target_sat,
+        organization=module_org,
+        location=module_location,
+        activation_keys=[module_activation_key.name],
+    )
+    assert result.status == 0, f'Failed to register host: {result.stderr}'
+    rhel_contenthost.execute('subscription-manager facts --update')
+    facts = rhel_contenthost.nailgun_host.get_facts(data={'per_page': 10000})['results'][
+        rhel_contenthost.hostname
+    ]
+    assert ip in facts.values()
```

Comment:
Instead I'd suggest to check the fact directly
```suggestion
    assert facts['net::interface::eth0::ipv4_address'] == ip
    assert facts['net::interface::eth0::mac_address'] == mac
```